sources/engine/Stride.Video/FFmpeg/FFmpegCodec.cs

```diff
@@ -159,77 +162,124 @@ public FFmpegCodec(GraphicsDevice graphcsDevice, AVCodecContext* originalContext
-                    contextd3d11->surface = (ID3D11VideoDecoderOutputView**)decoderOuputViewsHandle.Pointer;
-                    contextd3d11->surface_count = 1;
-
-                    pAVCodecContext->hwaccel_context = contextd3d11;
+                    unsafe
+                    {
+                        var iVideoContext = new ID3D11VideoContext { lpVtbl = (ID3D11VideoContextVtbl*)videoContext.Handle->LpVtbl };
+                        videoContextHandle = new PinnedObject<ID3D11VideoContext>(iVideoContext);
+                        contextd3d11->video_context = (ID3D11VideoContext*)videoContextHandle.Pointer;
+
+                        var iVideoDecoder = new ID3D11VideoDecoder { lpVtbl = (ID3D11VideoDecoderVtbl*)videoHardwareDecoder.Handle->LpVtbl };
+                        videoDecoderHandle = new PinnedObject<ID3D11VideoDecoder>(iVideoDecoder);
+                        contextd3d11->decoder = (ID3D11VideoDecoder*)videoDecoderHandle.Pointer;
+                        D3D11_VIDEO_DECODER_CONFIG* decoderConfigF = (D3D11_VIDEO_DECODER_CONFIG*) &decoderConfig;
+                        decoderConfigHandle = new PinnedObject<D3D11_VIDEO_DECODER_CONFIG>(*decoderConfigF);
+                        contextd3d11->cfg = (D3D11_VIDEO_DECODER_CONFIG*)decoderConfigHandle.Pointer;
+
+                        var iVideoOutputView = new ID3D11VideoDecoderOutputView { lpVtbl = (ID3D11VideoDecoderOutputViewVtbl*)videoHardwareDecoderView.Get().LpVtbl};
+                        decoderOuputViewsHandle = new PinnedObject<ID3D11VideoDecoderOutputView>(iVideoOutputView, true);
+                        contextd3d11->surface = (ID3D11VideoDecoderOutputView**)decoderOuputViewsHandle.Pointer;
+                        contextd3d11->surface_count = 1;
+                        pAVCodecContext->hwaccel_context = contextd3d11;
+
+                    }
+
+
                 }
             }
         }
+        private static Guid GetVideoDecoderProfile(Silk.NET.Direct3D11.ID3D11VideoDevice1 v, int i)
+        {
+            Guid profile;
+            unsafe
+            {
+                v.GetVideoDecoderProfile((uint)i, &profile);
+            }
+            return profile;
+        }
+
+        private static int CheckVideoDecoderFormat(Silk.NET.Direct3D11.ID3D11VideoDevice1 v,Guid p, Format f)
+        {
+            int i = 0;
+            unsafe
+            {
+                v.CheckVideoDecoderFormat(&p, f, &i);
+            }
+            return i;
+        }
 
-        private static IEnumerable<Guid> FindVideoFormatCompatibleProfiles(VideoDevice videoDevice, AVCodec codec)
+        private static IEnumerable<Guid> FindVideoFormatCompatibleProfiles(Silk.NET.Direct3D11.ID3D11VideoDevice1 videoDevice, AVCodec codec)
         {
-            for (var i = 0; i < videoDevice.VideoDecoderProfileCount; ++i)
+            for (var i = 0; i < videoDevice.GetVideoDecoderProfileCount(); ++i)
```

Comment:
This won't work as intended, and will likely crash.

In general, you should never have a struct representing an interface ever used by-value. 

i.e. `videoDevice` needs to be `ID3D11VideoDevice1*` (note the pointer) or `ComPtr<ID3D11VideoDevice1>`, not `ID3D11VideoDevice1`